pyscript.core/esm/runtime/_utils.js

```diff
@@ -1,10 +1,26 @@
 import { getBuffer } from "../fetch-utils.js";
-import { absoluteURL } from "../utils.js";
+import { absoluteURL, defineProperty } from "../utils.js";
+import "@ungap/with-resolvers";
+
+// REQUIRES INTEGRATION TEST
+/* c8 ignore start */
+// TODO: this should *NOT* be needed as the polyfill
+//       already patches on demand the Promise object
+const { withResolvers } = Promise;
+defineProperty(globalThis, "Promise", {
+    configurable: true,
+    value: class extends Promise {
+        withResolvers() {
+            return withResolvers.call(this);
+        }
+    },
+});
```

Comment:
What the heck does this mean? You get `withResolvers` from the existing `Promise` which has been polyfilled by `@ungap/with-resolvers` and then... write over `Promise` with a new subclass that has `withResolvers` as a non-static method, but `withResolvers` was supposed to be a static method??

Also, `Promise.withResolvers` looks really great, so many codebases have to make this utility one way or another.